packages/protocol/contracts/layer1/shasta/impl/Inbox.sol

```diff
@@ -149,19 +146,21 @@ abstract contract Inbox is EssentialContract, IInbox {
 
         // Verify capacity for new proposals
         uint256 availableCapacity = _getAvailableCapacity(config, coreState);
-        require(availableCapacity > 0, ExceedsUnfinalizedProposalCapacity());
+        require(availableCapacity > input.numForcedInclusions, ExceedsUnfinalizedProposalCapacity());
 
         // Process forced inclusion if required
-        bool forcedInclusionProcessed;
-        (coreState, forcedInclusionProcessed) = _processForcedInclusion(config, coreState);
-
-        if (!forcedInclusionProcessed || availableCapacity > 1) {
-            // Propose the normal proposal after the potential forced inclusion to match the
-            // behavior in Shasta fork.
-            LibBlobs.BlobSlice memory blobSlice =
-                LibBlobs.validateBlobReference(input.blobReference, _getBlobHash);
-            _propose(config, coreState, blobSlice, false);
+        coreState = _processForcedInclusions(config, coreState, input.numForcedInclusions);
+
+        // Verify that no forced inclusion that is due remains in the queue.
+        if (IForcedInclusionStore(config.forcedInclusionStore).isOldestForcedInclusionDue()) {
+            revert UnprocessedForcedInclusionIsDue();
         }
+
+        // Propose the normal proposal after the potential forced inclusion to match the
+        // behavior in Pacaya.
```

Comment:
imagine there is only one slot in the ring buffer, and there is a forced inclusion due, then we should still allow the next forced inclusion to be processed. Therefore, we need to change 

```
 require(availableCapacity > input.numForcedInclusions, ExceedsUnfinalizedProposalCapacity());
 ```

to 
```
 require(availableCapacity >= input.numForcedInclusions, ExceedsUnfinalizedProposalCapacity());
```

and wrap the regular proposal in a if-statement:


`if (availableCapacity > input.numForcedInclusions) {...}`